### DISCUSSION TITLE
How to clear assigned_object_type and assigned_object_id in IPAddress model using API

### DISCUSSION THREAD

**Discussion by AirDRoN-lab** [netbox-community/netbox#16900] [Help Wanted!]
Hello, everyone.
Not sure if this is a bug, but it possible.

Assigned_object_type and assigned_object_id in  IPAddress model doesn't clear (set null) through API requests. 

How to reproduce:
1) Have a Device with Interface A
2) Have an IP
3) Assign IP to Device Interface
4) 
```
curl --insecure -X 'PATCH' 'https://192.168.153.68/api/ipam/ip-addresses/13705/' \
  -H 'Accept: application/json' \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Token XXX' \
  -d '{
  "assigned_object_id": null
}'
```
5) 
Feedback from NetBox API:
```
{"assigned_object_id":["This field cannot be null."]}
```

So, how deassign IP from Device Interface? Need create an issue?
Similar discussion in #11454 but the problem is slightly different

Many thanks!

NetBox version 3.7.1

**Comment by ninabel** [comment]
I've used the 
"assigned_object_id": 0

But I see another problem - this doesn't change assigned_object_type.

**Reply by ninabel** [reply]
But it works when you set both fields to null:
        "assigned_object_id": null,
        "assigned_object_type": null,

**Comment by candlerb** [comment]
> Assigned_object_type and assigned_object_id in IPAddress model doesn't clear (set null) through API requests.

https://demo.netbox.dev/api/schema/swagger-ui/#/ipam/ipam_ip_addresses_partial_update

With Netbox version 4.1.4, it works if you set both assigned_object_id and assigned_object_type to null in the same request:

```
# curl -sSH "Authorization: Token $TOKEN" -H "Content-Type: application/json" -H  "Accept: application/json; indent=4" "$NETBOX/api/ipam/ip-addresses/524/" | grep assigned
    "assigned_object_type": "dcim.interface",
    "assigned_object_id": 24,
    "assigned_object": {

# curl -sSH "Authorization: Token $TOKEN" -H "Content-Type: application/json" -H  "Accept: application/json; indent=4" -XPATCH -d'{"assigned_object_type":null,"assigned_object_id":null}' "$NETBOX/api/ipam/ip-addresses/524/"
...

# curl -sSH "Authorization: Token $TOKEN" -H "Content-Type: application/json" -H  "Accept: application/json; indent=4" "$NETBOX/api/ipam/ip-addresses/524/" | grep assigned
    "assigned_object_type": null,
    "assigned_object_id": null,
    "assigned_object": null,
```
